cylc/flow/scheduler.py

```diff
@@ -642,6 +642,19 @@ async def run_scheduler(self) -> None:
                     self.restart_remote_init()
                     await commands.run_cmd(commands.poll_tasks(self, ['*/*']))
 
+                    # If we shut down with manually triggered waiting tasks,
+                    # submit them to run now.
+                    pre_prep_tasks = []
+                    for itask in self.pool.get_tasks():
+                        if (
+                            itask.is_manual_submit
+                            and itask.state(TASK_STATUS_WAITING)
+                        ):
+                            itask.waiting_on_job_prep = True
+                            pre_prep_tasks.append(itask)
+
+                    self.start_job_submission(pre_prep_tasks)
+
```

Comment:
I don't think it's right for tasks triggered with `cylc trigger --on-resume` to be run instantly if a workflow is restarted with `cylc play --pause`, however, this option won't last long and may only be used by a couple of people so we can handle some quirks.

Reply:
Yeah I was being a bit lazy there. I'll consider fixing that, just in case...